src/razor/src/document/razorDocumentManager.ts

```diff
@@ -219,6 +221,26 @@ export class RazorDocumentManager implements IRazorDocumentManager {
         this.notifyDocumentChange(document, RazorDocumentChangeKind.removed);
     }
 
+    private findDocument(path: string) {
+        // This method ultimately gets called from VS Code, using file system paths, or from DevKit, which
+        // can use paths as specified in the sln file. When these don't agree, on case-insensitive operating
+        // systems, we have to be careful to match things up correctly.
+
+        if (this.platformInfo.isLinux()) {
+            return this.razorDocuments[path];
+        }
+
+        const lowerCasePath = path.toLowerCase();
+        const key = Object.keys(this.razorDocuments).find(
```

Comment:
Can we store the doument path as lowercase to not have to do `toLowerCase` while searching? That seems like it could get expensive fairly quickly

Reply:
Looks like I can switch to a case insensitive comparison, rather than toLowering: https://github.com/dotnet/vscode-csharp/blob/main/src/razor/src/csharp/csharpProjectedDocumentContentProvider.ts#L76

I dare not poke at the bear that is "change the case of everything in the razorDocuments array" 😛